nemo/collections/common/tokenizers/canary_tokenizer.py

```diff
@@ -73,7 +73,53 @@ def text_to_ids(self, text, lang_id) -> list[int]:
             return self._tokenize_special_prompt(text)
         lang_id = _map_canary1_to_canary2_lang(lang_id, self.langs)
         if text.endswith(CANARY_EOS):
-            return super().text_to_ids(text[: -len(CANARY_EOS)], lang_id) + [self.eos_id]
+            text_no_eos = text[: -len(CANARY_EOS)]
+            time_pattern = re.compile(r"<\|\d+\|>")
+            time_text = "".join(time_pattern.findall(text_no_eos))
+            has_no_timestamp = not time_text
+            if has_no_timestamp:
+                return super().text_to_ids(text_no_eos, lang_id) + [self.eos_id]
+            else:
+                word_text = time_pattern.sub("", text_no_eos).strip()
```

Comment:
please refactor into a method `def _text_with_timestamps_to_ids(...)` and use `return self._text_with_timestamps_to_ids(...)` here

Reply:
Done.